pkgs/development/python-modules/pydantic-settings/default.nix

```
@@ -50,11 +50,21 @@ buildPythonPackage rec {
     export HOME=$TMPDIR
   '';
 
+  # We separate the tests from the package so that it does not depend on ruff.
+  # We do not want all of the downstream packages that depend on pydantic-settings to also depend on ruff.
+  doCheck = false;
+  passthru.tests = {
+    check = self.overridePythonAttrs {
```

```suggestion
    pytest = self.overridePythonAttrs {
```
is the convention I think.